index.js

```diff
@@ -104,8 +111,8 @@ Queue.prototype.createDuplexStream = function createDuplexStream (opts) {
       finish(self.inflight)
 
       function finish (inflight) {
-        debug('finish?', [self.pending, self.latestChange, inflight.since])
-        if (self.pending === 0 && self.latestChange === inflight.since) {
+        debug('finish?', [self.pending, self.latestChange, self.highestChange, inflight.since])
+        if (self.pending === 0 && self.highestChange === inflight.since) {
```

Comment:
looks good to me. i am trying to think of a case where this would be a false positive but I can't